src/table_properties.rs

```diff
@@ -0,0 +1,273 @@
+
+    pub fn raw_value_size(&self) -> u64 {
+        self.get_u64(DBTableProperty::RawValueSize)
+    }
+
+    pub fn num_data_blocks(&self) -> u64 {
+        self.get_u64(DBTableProperty::NumDataBlocks)
+    }
+
+    pub fn num_entries(&self) -> u64 {
+        self.get_u64(DBTableProperty::NumEntries)
+    }
+
+    pub fn format_version(&self) -> u64 {
+        self.get_u64(DBTableProperty::FormatVersion)
+    }
+
+    pub fn fixed_key_len(&self) -> u64 {
+        self.get_u64(DBTableProperty::FixedKeyLen)
+    }
+
+    pub fn column_family_id(&self) -> u64 {
+        self.get_u64(DBTableProperty::ColumnFamilyId)
+    }
+
+    pub fn column_family_name(&self) -> &str {
+        self.get_str(DBTableProperty::ColumnFamilyName)
+    }
+
+    pub fn filter_policy_name(&self) -> &str {
+        self.get_str(DBTableProperty::FilterPolicyName)
+    }
+
+    pub fn comparator_name(&self) -> &str {
+        self.get_str(DBTableProperty::ComparatorName)
+    }
+
+    pub fn merge_operator_name(&self) -> &str {
+        self.get_str(DBTableProperty::MergeOperatorName)
+    }
+
+    pub fn prefix_extractor_name(&self) -> &str {
+        self.get_str(DBTableProperty::PrefixExtractorName)
+    }
+
+    pub fn property_collectors_names(&self) -> &str {
+        self.get_str(DBTableProperty::PropertyCollectorsNames)
+    }
+
+    pub fn compression_name(&self) -> &str {
+        self.get_str(DBTableProperty::CompressionName)
+    }
+
+    pub fn user_collected_properties(&self) -> HashMap<&[u8], &[u8]> {
+        let mut res = HashMap::new();
+        unsafe {
+            let inner = crocksdb_ffi::crocksdb_table_properties_get_user_properties(self.inner);
+            let mut iter = UserCollectedPropertiesIter::new(self, inner);
+            while iter.valid() {
+                res.insert(iter.key(), iter.value());
+                iter.next();
+            }
+        }
+        res
+    }
+}
+
+struct UserCollectedPropertiesIter<'a> {
+    props: PhantomData<&'a TablePropertiesCollection>,
+    inner: *mut DBUserCollectedPropertiesIterator,
+}
+
+impl<'a> Drop for UserCollectedPropertiesIter<'a> {
+    fn drop(&mut self) {
+        unsafe {
+            crocksdb_ffi::crocksdb_user_collected_properties_iter_destroy(self.inner);
+        }
+    }
+}
+
+impl<'a> UserCollectedPropertiesIter<'a> {
+    fn new(_: &'a TableProperties,
+           inner: *mut DBUserCollectedProperties)
+           -> UserCollectedPropertiesIter<'a> {
+        unsafe {
+            UserCollectedPropertiesIter {
+                props: PhantomData,
+                inner: crocksdb_ffi::crocksdb_user_collected_properties_iter_create(inner),
+            }
+        }
+    }
+
+    fn valid(&self) -> bool {
+        unsafe { crocksdb_ffi::crocksdb_user_collected_properties_iter_valid(self.inner) }
+    }
+
+    fn next(&mut self) {
+        unsafe {
+            crocksdb_ffi::crocksdb_user_collected_properties_iter_next(self.inner);
+        }
+    }
+
+    fn key(&self) -> &'a [u8] {
```

Comment:
The returned `&'a [u8]` can live longer than the iterator, as long as the `TableProperties` is valid.